config-handler-router/src/test/groovy/org/codice/admin/router/ConfigurationHandlerRouterTest.groovy

```diff
@@ -0,0 +1,294 @@
+
+        then:
+        1 * configurationHandler.getCapabilities()
+        configurationHandler.getCapabilities().configurationType == CONFIGURATION_TYPE
+    }
+
+    def 'test configCapabilties(Request, Response) null config handler'() {
+        setup:
+        prepareConfigHandler(NON_MATCHING_ID, ConfigurationHandlerRouter.TEST_ID)
+
+        when:
+        def report = (Report) configurationHandlerRouter.configCapabilities(request, response)
+
+        then:
+        1 * response.status(400)
+        0 * configurationHandler.getCapabilities()
+        report.messages().size() == 1
+        report.messages().get(0).message() == FAILURE_MESSAGE
+        report.messages().get(0).configFieldId() == ConfigurationHandlerRouter.CONFIGURATION_TYPE_FIELD
+        report.containsFailureMessages()
+    }
+
+    def 'test configurations(Request, Response) success'() {
+        setup:
+        prepareConfigHandler(MATCHING_ID, ConfigurationHandlerRouter.TEST_ID)
+
+        when:
+        configurationHandlerRouter.configurations(request, response)
+
+        then:
+        1 * configurationHandler.getConfigurations()
+        configurationHandler.getCapabilities().configurationType == CONFIGURATION_TYPE
+    }
+
+    def 'test configurations(Request, Response) null config handler'() {
+        setup:
+        prepareConfigHandler(NON_MATCHING_ID, ConfigurationHandlerRouter.TEST_ID)
+
+        when:
+        def report = (Report) configurationHandlerRouter.configurations(request, response)
+
+        then:
+        1 * response.status(400)
+        0 * configurationHandler.configurations()
+        report.messages().size() == 1
+        report.messages().get(0).message() == FAILURE_MESSAGE
+        report.messages().get(0).configFieldId() == ConfigurationHandlerRouter.CONFIGURATION_TYPE_FIELD
+        report.containsFailureMessages()
+    }
+
+    def prepareConfigHandler(String configHandlerId, String id) {
+        response = Mock(Response)
+        configurationHandler = mockConfigurationHandler()
+        request = mockRequest(configHandlerId, id)
+
+        configurationHandlerRouter.setConfigurationHandlers([configurationHandler])
+    }
+
+    def mockConfigurationHandler() {
+        return Mock(ConfigurationHandler) {
+            getConfigurationHandlerId() >> MATCHING_ID
+            getConfigurationType() >> Mock(ConfigurationType) {
+                configClass() >> WfsSourceConfiguration
+                configTypeName() >> "sources"
+            }
+            getCapabilities() >> Mock(CapabilitiesReport) {
+                getConfigurationType() >> CONFIGURATION_TYPE
+            }
+        }
+    }
+
+    def mockRequest(String configHandlerId, String id) {
+        return Mock(Request) {
+            params(ConfigurationHandlerRouter.CONFIG_HANDLER_ID) >> configHandlerId
+            params(id) >> METHOD_ID
+            body() >> "{\"configurationType\":\"sources\",\"sourceHostName\":\"localhost\",\"sourcePort\":8993}"
```

Comment:
You could call the gson parser with a Configuration object instead of hard coding a string.